src/Database/Type/DateTimeType.php

```diff
@@ -206,9 +207,17 @@ public function marshal($value)
             $value['minute'],
             $value['second']
         );
-        $tz = isset($value['timezone']) ? $value['timezone'] : null;
+        $tz = isset($value['timezone']) ? $value['timezone'] : Configure::read('App.defaultOutputTimezone');
 
-        return new $class($format, $tz);
+        /* @var \Cake\I18n\Time $time */
+        $time = new $class($format, $tz);
+
+        $internalTimezone = date_default_timezone_get();
+        if ($tz && $tz !== $internalTimezone) {
+            $time = $time->timezone($internalTimezone);
+        }
```

Comment:
Does the time type need this kind of change as well?

Reply:
I would think so, so far I only checked datetime. If anyone has the chance to help with other types, that would be more than welcome.